bberg/src/verifier_builder.rs

```diff
@@ -47,21 +47,29 @@ impl VerifierBuilder for BBFiles {
         };
 
         let verify_proof_function_declaration: String = if has_public_input_columns {
-            format!("bool {name}Verifier::verify_proof(const HonkProof& proof, const std::vector<FF>& public_inputs)")
+            format!("bool {name}Verifier::verify_proof(const HonkProof& proof, const std::vector<std::vector<FF>>& public_inputs)")
         } else {
             format!("bool {name}Verifier::verify_proof(const HonkProof& proof)")
         };
 
-        let (public_inputs_check, evaluate_public_inputs) = if has_public_input_columns {
-            let public_inputs_column = public_cols[0].clone(); // asserted to be 1 for the meantime, this will be generalized when required
-            let inputs_check = format!(
+        let public_inputs_column_transformation = |public_inputs_column_name: &String, i: usize| {
+            format!(
                 "
-        FF public_column_evaluation = evaluate_public_input_column(public_inputs, circuit_size, multivariate_challenge);
-        if (public_column_evaluation != claimed_evaluations.{public_inputs_column}) {{
+        FF {public_inputs_column_name}_evaluation = evaluate_public_input_column(public_inputs[{i}], circuit_size, multivariate_challenge);
+        if ({public_inputs_column_name}_evaluation != claimed_evaluations.{public_inputs_column_name}) {{
             return false;
         }}
                 "
-            );
+            )
+        };
+
+        let (public_inputs_check, evaluate_public_inputs) = if has_public_input_columns {
+            let inputs_check = public_cols
+                .iter()
+                .enumerate()
+                .map(|(i, col_name)| public_inputs_column_transformation(&col_name, i))
+                .collect::<String>();
+
```

Comment:
Does not the code below be modified to support multiple columns? (e.g., evaluate_public_input_column )

Reply:
It is run per column, so its ok